pumpkin-core/src/random/mod.rs

```diff
@@ -1,10 +1,35 @@
+use std::{
+    sync::atomic::{AtomicU64, Ordering},
+    time,
+};
+
 use legacy_rand::{LegacyRand, LegacySplitter};
 use xoroshiro128::{Xoroshiro, XoroshiroSplitter};
 
 mod gaussian;
 pub mod legacy_rand;
 pub mod xoroshiro128;
 
+static SEED_UNIQUIFIER: AtomicU64 = AtomicU64::new(8682522807148012u64);
+
+pub fn get_seed() -> u64 {
+    let seed = SEED_UNIQUIFIER
+        .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |val| {
+            Some(val.wrapping_mul(1181783497276652981u64))
+        })
+        // We always return Some, so there will always be an Ok result
+        .unwrap();
+
+    let nanos = time::SystemTime::now()
+        .duration_since(time::SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_nanos();
+
+    let nano_upper = (nanos >> 8) as u64;
+    let nano_lower = nanos as u64;
+    seed ^ nano_upper ^ nano_lower
```

Comment:
You can/should use something similar to java's compareAndSet. You can look at https://github.com/kallekankaanpaa/jandom/blob/master/src/lib.rs#L228

Reply:
That's what fetch_update does I thought